pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -158,4 +158,484 @@ var _ = Describe("Bridge infrastructure configurator", func() {
+			queueCount         uint32
+			tapDeviceName      string
+			vmi                *v1.VirtualMachineInstance
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIBridgeInterface("default", "vm1")
+
+			bridgeConfigurator = NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+		})
+
+		BeforeEach(func() {
+			macStr := "AF:B3:1F:78:2A:CA"
+			mac, _ = net.ParseMAC(macStr)
+			mtu = 1000
+			podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: mtu}}
+			queueCount = uint32(0)
+			tapDeviceName = "tap0"
+		})
+
+		When("the pod features an L3 configured network (IPAM)", func() {
+			var (
+				dummySwap              *netlink.Dummy
+				podLinkAfterNameChange *netlink.GenericLink
+			)
+
+			BeforeEach(func() {
+				bridgeIPStr = fmt.Sprintf(bridgeFakeIP, 0)
+				bridgeIPAddr, _ = netlink.ParseAddr(bridgeIPStr)
+				dummySwap = &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Name: ifaceName}}
+				inPodBridge = &netlink.Bridge{LinkAttrs: netlink.LinkAttrs{Name: bridgeIfaceName}}
+				podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+				podLinkAfterNameChange = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: generateDummyIfaceName(ifaceName), MTU: mtu}}
+			})
+
+			BeforeEach(func() {
+				bridgeConfigurator.podNicLink = podLink
+				bridgeConfigurator.tapDeviceName = "tap0"
+				bridgeConfigurator.ipamEnabled = true
+				bridgeConfigurator.podIfaceIP = podIP
+			})
+
+			When("all network driver calls succeed", func() {
+				BeforeEach(func() {
+					handler.EXPECT().LinkSetDown(podLink).Return(nil)
+					handler.EXPECT().LinkAdd(inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetMaster(podLinkAfterNameChange, inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetUp(inPodBridge).Return(nil)
+					handler.EXPECT().ParseAddr(bridgeIPStr).Return(bridgeIPAddr, nil)
+					handler.EXPECT().AddrAdd(inPodBridge, bridgeIPAddr).Return(nil)
+					handler.EXPECT().DisableTXOffloadChecksum(inPodBridge.Name).Return(nil)
+					handler.EXPECT().CreateTapDevice(tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId).Return(nil)
+					handler.EXPECT().BindTapDeviceToBridge(tapDeviceName, inPodBridge.Name).Return(nil)
+					handler.EXPECT().LinkSetUp(podLinkAfterNameChange).Return(nil)
+					handler.EXPECT().LinkSetLearningOff(podLinkAfterNameChange).Return(nil)
+					handler.EXPECT().AddrDel(podLink, &podIP).Return(nil)
+					handler.EXPECT().LinkSetName(podLink, generateDummyIfaceName(ifaceName))
+					handler.EXPECT().LinkByName(generateDummyIfaceName(ifaceName)).Return(podLinkAfterNameChange, nil)
+					handler.EXPECT().LinkAdd(dummySwap).Return(nil)
+					handler.EXPECT().AddrReplace(dummySwap, &podIP).Return(nil)
+					handler.EXPECT().SetRandomMac(generateDummyIfaceName(ifaceName)).Return(mac, nil)
+					handler.EXPECT().ConfigureIpv4ArpIgnore().Return(nil)
```

Comment:
We can remove "Return(nil)" since it's implici behaviour by gomock, similar to "Times(1)"

Reply:
Did not know that.